src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -243,7 +243,7 @@ private async Task<Uri> BuildAuthorityWithRegionAsync(Uri canonicalAuthority, Re
 
                     s_region = regionToUse;
                     requestContext.ApiEvent.FallbackToGlobal = false;
-                    requestContext.Logger.Info($"[Region discovery] Region auto detection failed. Region provided by the user will be used: ${regionToUse}.");
+                    requestContext.Logger.Info($"[Region discovery] Region auto detection failed. Region provided by the user will be used: {regionToUse}.");
```

Comment:
There's potentially a null ref issue at line 215 - 

```if (string.IsNullOrEmpty(str))``` is the correct null check.